finetuner/data.py

```diff
@@ -58,6 +63,28 @@ def build_dataset(
     return data
 
 
+def build_encoding_dataset(
+    model: Union['InferenceEngine', str], data: Union[List[str], DocumentArray]
+) -> DocumentArray:
+    """If data has been provided as a list, a :class:`DocumentArray` is created
+    from the elements of the list
+    """
+    if not isinstance(data, DocumentArray):
+        try:
+            task = get_stub(model._model_name, select_model=model._select_model).task
+            if task == 'any':
+                raise ValueError('MLP model does not support values from a list.')
+            task = task.split('-to-')[0]
+        except Exception:
+            task = model._model_name.split('-')[1]
```

Comment:
I think here is something wrong. When a clip model is used, you can encode text as well as images with it. So it can either be text or image and you can not infer this from the task. Moreover, I don't understand what the except block does. If you pass any string to `test`, `test.split('-to-')[0]` will always just return the text.